### DISCUSSION TITLE
ES5 Compatibility build for outdated bundler ?

### DISCUSSION THREAD

**Discussion by stevebeauge** [rayepps/radash#185] [Q&A]
Hi,

Starting from v9, radash stopped to work in my corporate application. After some (non obvious) investigations, I discovered that the bundler of the application cannot import ES2020 modules (at least some of the new syntax, like `someArray?.length`).

Because the bundler (which is webpack 4) is actually burried behind 3rd party application, I cannot change the config easily.

I won't ask the radash team to revert to older module format (looking forward, not backward), but I wonder if there's any workaround to get radash as ES5 module.

Or if the rollup config can add a new output "legacy ESM" aside the more modern ones.

Thanks for insights

**Comment by matteosacchetto** [comment]
Hi there,
wait, there is a range of options in between ES5 and ES2020, where you have ES6 modules, but with less modern syntax.

I will think something to address this and share some possible ideas as soon as I have something which could work.
In the meantime, I suggest you stick with the latest version which worked

Cheers

**Reply by matteosacchetto** [reply]
One additional thing: could you provide some details about which node version and package manager version you are using?

**Reply by stevebeauge** [reply]
Hello @matteosacchetto 

I mostly work with pnpm 7 and node v16. But I'm not sure to get the reason of your question. My actual issue is related to webpack 4. When I use radash with a "simple" typescript compilation, I get no error. But as as soon as I try to bundle using my webpack config, it fails.

Actually I have no control (or very few) over webpack config because I'm building addons for a commercial product that ships with a bundler.

Thx
